### PR TITLE
[Wave 6: Referral Program] Promote referral program messaging in NewDot Redesign the Referral page

### PR DESCRIPTION
**Pull request by rezkiy37**
### Details

The PR redesigns a page for Referral Program. Currently, it has 5 variants. For 2 (_refer a friend, share code_) of them, we changed content.

### Fixed Issues

$ https://github.com/Expensify/App/issues/32187
PROPOSAL: N/A

#### Money request
1. Click on the **FAB**.
3. Click on **"Request money"**
4. Enter any amount.
5. Click on **"Next"**.
6. Click on **"Request money, get $250."**.
7. Verify that the page looks correctly.

#### Start chat
1. Click on the **FAB**.
2. Click on **"Start a chat"**
5. Click on **"Start a chat, get $250.**.
6. Verify that the page looks correctly.

#### Send money
1. Click on the **FAB**.
2. Click on **"Send money"**
3. Enter any amount.
4. Click on **"Next"**.
5. Click on **"Send money, get $250."**.
6. Verify that the page looks correctly.

#### Refer a friend
1. Click on the **Search**.
2. Click on **"Refer a friend, get $250."**.
3. Verify that the page looks correctly.
4. Click on **"Copy referral link"**.
5. Verify that the button label has been changed to "**Copied!**" ("**¡Copiado!**") for a moment.
6. Verify that you actually copied a link - `https://join.my.expensify.com/?thanks=<YOUR_LOGIN>`.

#### Share code
1. Go to **Settings**.
2. Click on **"Share code"**
3. Click on **"Get $250"**.
5. Verify that the page looks correctly.
6. Click on **"Copy referral link"**.
8. Verify that the button label has been changed to "**Copied!**" ("**¡Copiado!**") for a moment.
9. Verify that you actually copied a link - `https://join.my.expensify.com/?thanks=<YOUR_LOGIN>`.

##### "**Learn more**" click
1. Click on "**Learn more**" button on a page.
2. Verify that it redirects you to `https://help.expensify.com/articles/new-expensify/billing-and-plan-types/Referral-Program`.

##### "**Learn more**" long press/right-click with mouse
1. Long press (_on mobile devices_) / right-click with mouse (_on PC_) on "**Learn more**" button on a page.
2. Verify that a context menu with the link appears.

- [X] Verify that no errors appear in the JS console

### Offline tests

Same as "**Tests**".

### QA Steps

Same as "**Tests**".

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Android Eng Learn more
Android Eng Refer a friend
Android Eng Request money
Android Eng Send money
Android Eng Share code
Android Eng Start a chat
Android ES Learn more
Android ES Refer a friend
Android ES Request money
Android ES Send money
Android ES Share code
Android ES Start a chat

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Android Chrome Eng Learn more
Android Chrome Eng Refer a friend
Android Chrome Eng Request money
Android Chrome Eng Send money
Android Chrome Eng Share code
Android Chrome Eng Start a chat
Android Chrome ES Learn more
Android Chrome ES Refer a friend
Android Chrome ES Request money
Android Chrome ES Send money
Android Chrome ES Share code
Android Chrome ES Start a chat

</details>

<details>
<summary>iOS: Native</summary>

IOS Eng Learn more
IOS Eng Refer a friend
IOS Eng Request money
IOS Eng Send money
IOS Eng Share code
IOS Eng Start a chat
IOS ES Learn more
IOS ES Refer a friend
IOS ES Request money
IOS ES Send money
IOS ES Share code
IOS ES Start a chat

</details>

<details>
<summary>iOS: mWeb Safari</summary>

IOS Safari Eng Learn more
IOS Safari Eng Refer a friend
IOS Safari Eng Request money
IOS Safari Eng Send money
IOS Safari Eng Share code
IOS Safari Eng Start a chat
IOS Safari ES Learn more
IOS Safari ES Refer a friend
IOS Safari ES Request money
IOS Safari ES Send money
IOS Safari ES Share code
IOS Safari ES Start a chat

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Chrome Eng Learn more
Chrome Eng Refer a friend
Chrome Eng Request money
Chrome Eng Send money
Chrome Eng Share code
Chrome Eng Start a chat
Chrome ES Learn more
Chrome ES Refer a friend
Chrome ES Request money
Chrome ES Send money
Chrome ES Share code
Chrome ES Start a chat

</details>

<details>
<summary>MacOS: Desktop</summary>

Desktop Eng Learn more
Desktop Eng Refer a friend
Desktop Eng Request money
Desktop Eng Send money
Desktop Eng Share code
Desktop Eng Start a chat
Desktop ES Learn more
Desktop ES Refer a friend
Desktop ES Request money
Desktop ES Send money
Desktop ES Share code
Desktop ES Start a chat

</details>

### CODE REVIEW

**Comment by rezkiy37** [comment]
@jamesdeanexpensify, please verify that everything looks good and content is correct. 

<details><summary>Share code</summary>

Screenshot 2023-12-02 at 15 43 41

</details> 

<details><summary>Refer a friend</summary>

Screenshot 2023-12-02 at 15 43 44

</details> 

<details><summary>Request money</summary>

Screenshot 2023-12-02 at 15 43 47

</details> 


<details><summary>Start a chat</summary>

Screenshot 2023-12-02 at 15 43 50

</details> 

<details><summary>Send money</summary>

Screenshot 2023-12-02 at 15 43 52

</details>

**Comment by jamesdeanexpensify** [comment]
I'm going to share these internally today for any final feedback in case we're missing anything. Will be back soon, thanks!

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/pages/ReferralDetailsPage.js [hunk 1]
```diff
@@ -40,80 +42,62 @@ const defaultProps = {
...
-                    style={[styles.w100]}
-                    text={translate('common.buttonConfirm')}
-                    onPress={() => Navigation.goBack(getFallbackRoute())}
-                    pressOnEnter
-                    enterKeyEventListenerPriority={1}
+            }
+            headerContainerStyles={[styles.staticHeaderImage, styles.justifyContentCenter]}
+            backgroundColor={theme.PAGE_THEMES[SCREENS.SETTINGS.WORKSPACES].backgroundColor}
+        >
+            <Text style={[styles.textHeadline, styles.textAlignCenter, styles.mb3, styles.mt8, styles.ph4]}>{contentHeader}</Text>
```
[the remark this answers is not in the document]
Comment (rezkiy37) [reply src/pages/ReferralDetailsPage.js:75]: So, it is already has bottom margins. Why do we need paddings?

File: src/pages/ReferralDetailsPage.js [hunk 2]
```diff
@@ -40,80 +42,62 @@ const defaultProps = {
...
-                <Button
-                    success
-                    style={[styles.w100]}
-                    text={translate('common.buttonConfirm')}
-                    onPress={() => Navigation.goBack(getFallbackRoute())}
-                    pressOnEnter
-                    enterKeyEventListenerPriority={1}
+            }
+            headerContainerStyles={[styles.staticHeaderImage, styles.justifyContentCenter]}
+            backgroundColor={theme.PAGE_THEMES[SCREENS.SETTINGS.WORKSPACES].backgroundColor}
```
[the remark this answers is not in the document]
Comment (rezkiy37) [reply src/pages/ReferralDetailsPage.js]: Done - 0d31230.

**Comment by rezkiy37** [comment]
Fixed UI bugs.

Screenshot 2023-12-07 at 19 21 52

File: src/pages/ReferralDetailsPage.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rezkiy37) [reply src/pages/ReferralDetailsPage.js]: Done - 3870b4d.

Screenshot 2023-12-07 at 19 37 29

**Comment by jamesdeanexpensify** [comment]
We're still making the button "Copy invite link" right? Thanks!

**Comment by grgia** [comment]
@parasharrajat it should be 32px

**Comment by rezkiy37** [comment]
1. Changed title of the copy button.
2. Updated spacing between the top block and content.

Screenshot 2023-12-08 at 10 56 28

Screenshot 2023-12-08 at 10 57 55

**Review by parasharrajat — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

:ribbon: :eyes: :ribbon:  C+ reviewed

File: src/languages/es.ts [hunk 3]
```diff
@@ -2442,31 +2442,30 @@ export default {
...
         },
         [CONST.REFERRAL_PROGRAM.CONTENT_TYPES.SHARE_CODE]: {
             buttonText1: `Recibe $${CONST.REFERRAL_PROGRAM.REVENUE}`,
-            header: `Recibe $${CONST.REFERRAL_PROGRAM.REVENUE} por cada recomendación`,
-            body1: 'Si conoces a alguien que dedique demasiado tiempo a los gastos (literalmente cualquiera: tu vecino, tu jefe, tu amigo de contabilidad), envíale tu enlace de invitación de Expensify:',
-            body2: `Cuando comiencen una suscripción anual, obtendrás $${CONST.REFERRAL_PROGRAM.REVENUE}. Así de fácil.`,
+            header: `Invita a un amigo y obtiene $${CONST.REFERRAL_PROGRAM.REVENUE}`,
+            body: `Sé el primero en invitar a un amigo (o a cualquier otra persona) a Expensify y obtiene $${CONST.REFERRAL_PROGRAM.REVENUE} si se convierte en cliente de Expensify. Comparte tu enlace de invitación por SMS, email o publícalo en las redes sociales.`,
         },
         copyReferralLink: 'Copiar enlace de invitación',
```
[the remark this answers is not in the document]
Comment (rezkiy37) [reply src/languages/es.ts:2470]: @parasharrajat, yes, it is.

File: src/pages/ReferralDetailsPage.js [hunk 4]
```diff
@@ -40,80 +42,62 @@ const defaultProps = {
...
+                    icon={Expensicons.Copy}
+                    successIcon={Expensicons.Checkmark}
+                    successText={translate('qrCodes.copied')}
+                    onPress={() => Clipboard.setString(referralLink)}
                 />
-            </FixedFooter>
-        </ScreenWrapper>
+            )}
+
+            <MenuItem
```
[the remark this answers is not in the document]
Comment (hayata-suenaga) [reply src/pages/ReferralDetailsPage.js:89]: yep I agree with @rezkiy37 👍

**Review by hayata-suenaga — DISMISSED** [review]
LGTM

**Comment by rezkiy37** [comment]
The conflicts were resolved.

**Review by hayata-suenaga — DISMISSED** [review]
good work! LGTM!

**Comment by hayata-suenaga** [comment]
there are conflicts 😢

**Comment by rezkiy37** [comment]
@hayata-suenaga, resolved.

**Comment by rezkiy37** [comment]
A bug was found. Investigating...
Screenshot 2023-12-12 at 10 36 53

**Comment by rezkiy37** [comment]
Fixed.
Screenshot 2023-12-12 at 10 40 19

**Review by hayata-suenaga — APPROVED** [review]
thanks for fixing the merge conflicts 👍

**Comment by jamesdeanexpensify** [comment]
Actually, just kidding. It works! Sorry bout that.

**Pull request merged**